src/input/process.rs

```diff
@@ -265,4 +303,91 @@ mod tests {
     fn test_create_process_map_missing_parameters() {
         test_missing!(parameters);
     }
+
+    #[test]
+    fn test_validate_commodities() {
+        // Create mock commodities
+        let commodity_sed = Rc::new(Commodity {
+            id: "commodity_sed".into(),
+            description: "SED commodity".into(),
+            kind: CommodityType::SupplyEqualsDemand,
+            time_slice_level: TimeSliceLevel::Annual,
+            costs: CommodityCostMap::new(),
+            demand: DemandMap::new(),
+        });
+
+        let commodity_non_sed = Rc::new(Commodity {
+            id: "commodity_non_sed".into(),
+            description: "Non-SED commodity".into(),
+            kind: CommodityType::ServiceDemand,
+            time_slice_level: TimeSliceLevel::Annual,
+            costs: CommodityCostMap::new(),
+            demand: DemandMap::new(),
+        });
+
+        let commodities: HashMap<Rc<str>, Rc<Commodity>> = vec![
```

Comment:
Could just be a slice cf. a `Vec`:

```suggestion
        let commodities: HashMap<Rc<str>, Rc<Commodity>> = [
```